src/user/wikis/Wikis.js

```diff
@@ -0,0 +1,352 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import "./wikis.scss";
+import axios from 'axios';
+import Page from "./Page/Page";
+import Layout from "./Layout/Layout";
+import Editor from "./Editor/Editor";
+import Sidebar from "./Sidebar/Sidebar";
+import { BASE_URL } from "../../actions/baseApi";
+import { getWikis } from "../../actions/wikisAction";
+import { ToastContainer, toast } from "react-toastify";
+
+class Wikis extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      wikis: true,
+      allWikis: [],
+      currentPage: 1,
+      editorMode: false,
+      historyMode: false,
+      viewHistory: false,
+      newPageEditor: false,
+      sidebarEditor: false,
+      spinner: "Loading...",
+    };
+    this.axiosCancel = axios.CancelToken.source();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({
+      allWikis: nextProps.wikis.wikis,
+      spinner: "",
+    });
+  }
+
+  componentDidMount() {
+    setTimeout(() => {
+      this.props.getWikis();
+    });
+  }
+
+  handleEditorMode = (newPage = false, sidebar = false) => {
+    this.setState({
+      editorMode: true,
+      newPageEditor: !!newPage,
+      sidebarEditor: !!sidebar,
+    });
+  };
+
+  cancelEditor = () => {
+    this.setState({
+      editorMode: false,
+      newPageEditor: false,
+    });
+  };
+
+  setView = async (page, ref) => {
+    let pos = 0;
+    this.state.allWikis.forEach((ele, index) => {
+      if (ele.title === page) pos = index;
+    });
+    if (pos) {
+      this.setState(
+        {
+          spinner: "Switing Page.... ",
+        },
+        async () => {
```

Comment:
I think you can move these server request into actions

Reply:
Done